packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -210,4 +204,57 @@ const Separator = ({ separator }: Pick<BreadcrumbsProps, 'separator'>) => (
   </Box>
 );
 
+const BreadcrumbsItem = React.forwardRef((props, forwardRef) => {
+  const { as, children, className, ...rest } = props;
+
+  let itemType = 'span';
+
+  if (as) {
+    itemType = as;
+  } else if (!!props.href) {
+    itemType = 'a';
+  } else if (!!props.onClick) {
+    return (
+      <Box
+        as={'button' as 'a'}
+        className={cx('iui-breadcrumbs-button', className)}
+        ref={forwardRef}
+        {...rest}
+      >
+        <Box as='span' className={cx('iui-breadcrumbs-text', className)}>
```

Comment:
any chance we can get rid of this extra span? 

it should be possible to use the same single class for both anchors and buttons, because we already do this in other places (such as iui-anchor and iui-alert-link and iui-toast-anchor).

Reply:
Each of the examples you've listed (iui-anchor, iui-alert-link, and iui-toast-anchor) do not apply different styling for button vs anchor. Should iui-breadcrumbs-action have the same styling for both button and anchor elements?

Reply:
it is very similar though.
![](https://github.com/iTwin/iTwinUI/assets/9084735/01b3142c-4abf-4ea0-a961-609cbe3ef7db)

for the parts that are different, you can apply specific styles targeting `a` or `button`. e.g.:
```scss
.iui-breadcrumbs-action {
  // ...shared styles

  &:where(a) {
    // ...anchor-specific
  }

  &:where(button) {
    // ...button-specific
  }
}
```

both anchor and toast do this to a lesser extent.

https://github.com/iTwin/iTwinUI/blob/92f81adaf94d6874b710de026606cddc779a443e/packages/itwinui-css/src/anchor/mixins.scss#L5-L6

https://github.com/iTwin/iTwinUI/blob/92f81adaf94d6874b710de026606cddc779a443e/packages/itwinui-css/src/toast/toast.scss#L153-L154

Reply:
This has been updated. I had to use `display: inline-block` for the `<button class='iui-breadcrumbs-action'>Previous</button>` element since there is no way to set `text-overflow` on a `button` element that has `display: flex` (to my knowledge). As a result, the icon for the overflow button has been shifted up slightly

![image](https://github.com/iTwin/iTwinUI/assets/78564221/53b18b68-bb56-462f-b048-0790dd880a52)

My suggestion on how to fix this is adding an extra class name or data attribute to `iui-breadcrumbs-action` called `iui-empty` which is true if the button doesn't have any children elements besides plain text. Is there a better solution for this?

Reply:
How it normally works in a button is it uses a `<span>`	inside the `<button>` for the text. Maybe we should use the same setup? (We could even reuse `<Button styleType='borderless'>` since it's already set up, but that would get more complicated.)

Reply:
This has been changed. `<span>` is used inside of `<button>` for the button text, but its original class name of `iui-breadcrumbs-text` has been removed

Reply:
Looks much better with the shared class between `a` and `button`! I wish we could avoid the extra span, but it's a necessary evil because of the overflow, so it's ok - at least it doesn't have a class. I have left another suggestion in https://github.com/iTwin/iTwinUI/pull/1369#discussion_r1247032801